File: 0058-length-of-last-word/0058-length-of-last-word.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        int length = 0, i = s.length() - 1;
        while (i >= 0) {
            if (s[i] == ' ') {
                if (length == 0) --i;
                else return length;
            }
            
            else {
                --i;
                ++length;
            }
        }
        return length;
    }
};

File: 0058-length-of-last-word/0058-length-of-last-word.py
class Solution(object):
    def lengthOfLastWord(self, s):
        """
        :type s: str
        :rtype: int
        """
        i, length = len(s) - 1, 0
        while s[i] == ' ':
                i -= 1
        while i >= 0:
            if s[i] == ' ':
                return length
            else:
                length += 1
                i -= 1
        return length